### PR TITLE
Alternative implementation of the meta wrapper

### PR DESCRIPTION
**Pull request by mcollina**
@jsumners @davidmarkclements let me know

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -249,3 +238,39 @@ module.exports = function prettyFactory (options) {
...
+    write (chunk) {
+      var chindings = this.lastLogger[parsed]
+
+      if (!chindings) {
+        chindings = JSON.parse('{"v":1' + this.lastLogger.chindings + '}')
+        this.lastLogger[parsed] = chindings
+      }
+
+      // TODO remove Object.assign as its slow
+      const obj = Object.assign({
```
Comment (jsumners) [inline index.js]: I tend to use https://www.npmjs.com/package/conflate when I'm looking for something other than `Object.assign`. See https://github.com/jsumners/merge-benchmarks

File: test/basic.test.js [hunk 2]
```diff
@@ -354,7 +354,7 @@ test('basic prettifier tests', (t) => {
         cb()
       }
     })
-    const prettyStream = prettyFactory({useMetadata: true, outputStream: dest})
+    const prettyStream = prettyFactory().asMetaWrapper(dest)
```
Comment (jsumners) [inline test/basic.test.js]: I'm confused here. How will options like `colorize` be recognized?

Comment (jsumners) [reply test/basic.test.js]: Oh, I see how it works now. Can you add a simple unit test to show options being passed in? I think `levelFirst` would be sufficient.

File: index.js (hunk 1 above)
Comment (mcollina) [reply index.js]: probably we should just leave `Object.assign`, the performance is not that horrible compared to others.

Comment (jsumners) [reply index.js]: I am 👌 with `Object.assign`.

**Comment by mcollina** [comment]
Updated.

**Comment by jsumners** [comment]
Just missing the readme update.

**Pull request merged**